WordPress/src/main/java/org/wordpress/android/viewmodel/pages/ActionPerformer.kt

```diff
@@ -1,25 +1,26 @@
 package org.wordpress.android.viewmodel.pages
 
+import kotlinx.coroutines.experimental.launch
 import org.greenrobot.eventbus.Subscribe
 import org.greenrobot.eventbus.ThreadMode
 import org.wordpress.android.fluxc.Dispatcher
-import org.wordpress.android.fluxc.store.PostStore.OnPostUploaded
-import javax.inject.Inject
 import org.wordpress.android.fluxc.model.CauseOfOnPostChanged
 import org.wordpress.android.fluxc.store.PostStore.OnPostChanged
+import org.wordpress.android.fluxc.store.PostStore.OnPostUploaded
 import org.wordpress.android.util.coroutines.suspendCoroutineWithTimeout
 import org.wordpress.android.viewmodel.pages.ActionPerformer.PageAction.EventType
 import org.wordpress.android.viewmodel.pages.ActionPerformer.PageAction.EventType.DELETE
 import org.wordpress.android.viewmodel.pages.ActionPerformer.PageAction.EventType.UPDATE
 import org.wordpress.android.viewmodel.pages.ActionPerformer.PageAction.EventType.UPLOAD
+import javax.inject.Inject
 import kotlin.coroutines.experimental.Continuation
 
 class ActionPerformer
 @Inject constructor(private val dispatcher: Dispatcher) {
-    private var continuations: MutableMap<Long, Map<EventType, Continuation<Boolean>>> = mutableMapOf()
+    private var continuations: MutableMap<Pair<Long, EventType>, Continuation<Pair<Boolean, Long>>> = mutableMapOf()
 
     companion object {
-        private const val ACTION_TIMEOUT = 30L * 1000
+        private const val ACTION_TIMEOUT = 10L * 1000
```

Comment:
Why was this changed from 30 seconds to 10 seconds?

Reply:
I realized that 30s may seem too long when there's something going on with the network and nothing's happening visually. For a page update, 10s should be enough and if it's not the connection is probably wonky.